### PR TITLE
fix: remove fieldset from custom demo, plus other fixes

### LINKED ISSUE #32749: Single SELECT "labeled" with a FIELDSET and LEGEND? (bad or even invalid use for FIELDSET?)
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/select

### What specific section or headline is this issue about?

Customizing select styles

### What information was incorrect, unhelpful, or incomplete?

Under [this section](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/select#customizing_select_styles) (heading, <q>Customizing select styles</q>), the HTML model used is a single `select` (no other interactive elements) inside a `fieldset`. Nowhere else on the page is the `select` "labeled" with a `fieldset` + `legend`, which is great, since the [HTML Living spec](https://html.spec.whatwg.org/multipage/form-elements.html#the-fieldset-element) notes `fieldset` is for a <q> set of form controls</q> (plural):

> The `fieldset` element represents a set of form controls (or other content) grouped together, optionally with a caption. The caption is given by the first `legend` element that is a child of the `fieldset` element, if any. The remainder of the descendants form the group.

[This MDN page](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/fieldset) also notes `fieldset` are to <q>group several controls</q> (again, plural):

> The `fieldset` HTML element is used to group several controls as well as labels (`label`) within a web form.

So is it even valid, let alone exemplary, that a **single** `select` is *effectively* being labeled by a `fieldset` (+`legend`)?

Shouldn't it be labeled with a `label` as it is elsewhere on the page?

I assume it is effectively invalid to use `fieldset` (+`legend`) to contain a single interactive element, given the wording of the spec' (and the wording in the MDN stated page)?

### What did you expect to see?

`label` not `fieldset` used as the label in the code example.

### Do you have any supporting links, references, or citations?

See above.

### Do you have anything more you want to share?

I use these docs to demonstrate best practices and I either want to understand how the code used at present is a best practice, and learn from it, or, if I am correct and it is poor, I would love the page to be changed so I can continue to link to it to demonstrate how `select` is used and not "pollute" that demo by people seeing it being "wrongly" labeled.

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/html/element/select`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/select
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/html/element/select/index.md
* Last commit: https://github.com/mdn/content/commit/fdd3ac5598c3ddceb71e59949b003936ae99f647
* Document last modified: 2024-02-22T19:57:33.000Z

</details>

### PR DESCRIPTION
**Pull request by chrisdavidmills**
### Description

As per https://github.com/mdn/content/issues/32749, the `<select>` demo [Customizing select styles](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/select#customizing_select_styles) shouldn't really use `<fieldset>`/`<legend>` to label a single `<select>`. There were other issues too.

This PR:

- Replaces `<fieldset>`/`<legend>` with `<section>`/`<label>`.
- Gives the `<select>` elements different `id` values, so we don't have duplicate IDs.
- Removes the `required` and `autocomplete` attributes, as they don't seem needed at all and distract from the point of the demo.
- Adds an `<hr>` and a bit of padding to separate things out a bit.
- Cuts down on the amount of selector cruft in the CSS — the selector chains before were way over the top.

I didn't:

- Fix the issue with wonky taxindexing (the custom select is tabbed to first, then the standard one) due to how taxindex is used to allow the custom select keyboard accessibility to work. I ran out of time on this, and I didn't consider it vital to fix. If anything, it acts as a reminder of the dangers of trying to reimplement browser form behavior yourself.

### Related issues and pull requests

Fixes https://github.com/mdn/content/issues/32749

### CODE REVIEW

File: files/en-us/web/html/element/select/index.md [hunk 1]
```diff
@@ -369,52 +379,40 @@ html body form fieldset#custom div.select datalist {
...
   padding: 0.5em;
   border-width: 0;
 }
 
-html body form fieldset#custom div.select datalist div.option:hover,
-html body form fieldset#custom div.select datalist div.option:focus,
-html body form fieldset#custom div.select datalist div.option:checked {
+section#custom div.select datalist div.option:hover,
+section#custom div.select datalist div.option:focus,
+section#custom div.select datalist div.option:checked {
```
Comment (estelle) [inline files/en-us/web/html/element/select/index.md:392]:
```suggestion
.select datalist .option:hover,
.select datalist .option:focus,
.select datalist .option:checked {
```
this is not your doing, but it is not good code. Datalists generally only contain options, though the spec reads "Either: [phrasing content](https://html.spec.whatwg.org/multipage/dom.html#phrasing-content-2).
    Or: Zero or more [option](https://html.spec.whatwg.org/multipage/form-elements.html#the-option-element) and [script-supporting](https://html.spec.whatwg.org/multipage/dom.html#script-supporting-elements-2) elements." 
    
My preference is to get rid of this entire example. If we keep it, we should create a component `<template>`, `<slot>`, etc.If we do keep it, the CSS selectors all needs to edited down.

**Review by estelle — CHANGES_REQUESTED** [review]
We should either remove this example, or, if we keep it, rewrite is as an HTML custom element and demonstrate best practices for using `<template>`, `<slot>`, ::part, etc.

The basis for the edit is valid and an improvement on what was there, but the original has the worst selectoritis I've seen on MDN, and the edit replaces rather than improves it. Let's improve it if we keep this.

File: files/en-us/web/html/element/select/index.md (hunk 1 above)
Comment (estelle) [reply files/en-us/web/html/element/select/index.md:392]: We can now style select with CSS (no JS). See https://codepen.io/5t3ph/pen/MWyyYNz

**Comment by chrisdavidmills** [comment]
@estelle thanks for your review. I've accepted all your changes — it is good to further improve on those selectors.

As for what to do next — I don't have time to significantly rewrite the example to use web components, and I don't want to delete the example either. Flawed as it is, I think it is still useful to some degree. I also feel that I have fixed the issue that was filed, and these suggestions are out of scope for it.

**Review by estelle — APPROVED** [review]

**Pull request merged**